src/overlays/actors/ovl_En_Wdhand/z_en_wdhand.c

```diff
@@ -105,63 +175,679 @@ static DamageTable D_80AF64F4 = {
+            } else {
+                this->limbRotations[0].y += (s16)(s32)Rand_CenteredFloat(12288.0f);
+            }
+            this->limbRotations[i].z = Rand_S16Offset(0x2000, 0x1000);
+        }
+        t += 2;
+    }
+
+    if (this->timer < 4) {
+        Math_StepToF(&this->globalLimbScaleFactor, 1.5f, 1.0f / 8);
+    } else if (this->timer < 16) {
+        Math_StepToF(&this->globalLimbScaleFactor, 2.0f, 1.0f / 24);
+    }
+
+    if (this->timer == 0) {
+        this->collider.base.ocFlags1 |= OC1_ON;
+        EnWdhand_SetupReturnToIdle(this);
+    } else if (this->timer >= 4) {
+        Vec3f limbPos;
+
+        // Create a model -> world matrix
+        Matrix_Put(&this->relativeToWorldTransform);
+        Matrix_Scale(0.01f, 0.01f, 0.01f, MTXMODE_APPLY);
+
+        Matrix_MultVecY(300.0f, &limbPos);
+
+        for (i = 0; i < EN_WDHAND_NUM_SEGMENTS; i++) {
+            Matrix_Push();
+            // Further rotate and translate
+            EnWdhand_SetTransform(this, i, false, &limbPos);
+            Matrix_MultVecY(2300.0f, &limbPos);
+            Matrix_Pop();
+        }
+
+        // Set transform for the hand
+        EnWdhand_SetTransform(this, EN_WDHAND_NUM_SEGMENTS, true, &limbPos);
+        // Copy the hand rotation to player shape rotation
+        Matrix_MtxFToYXZRot(Matrix_GetCurrent(), &player->actor.shape.rot, false);
+
+        if (this->globalLimbScaleFactor > 1.0f) {
+            f32 zRotFactor = CLAMP_MAX(this->globalLimbScaleFactor, 1.5f);
+
+            Matrix_RotateZS((zRotFactor - 1.0f) * -32768.0f, MTXMODE_APPLY);
+        }
+        Matrix_MultVecY(1000.0f, &player->actor.world.pos);
+
+        if ((this->timer == 4) && (player->actor.parent == &this->actor)) {
+            // Throws the player, releasing them
+            player->av2.actionVar2 = 100;
+            player->actor.parent = NULL;
+            // Transpose again
+            Matrix_Transpose(&this->relativeToWorldTransform);
+            player->actor.shape.rot.x = 0;
+            player->actor.shape.rot.z = 0;
+            player->actor.world.pos.x += 2.0f * this->actor.velocity.x;
+            player->actor.world.pos.y += 2.0f * this->actor.velocity.y;
+            player->actor.world.pos.z += 2.0f * this->actor.velocity.z;
+            func_800B8D50(play, &this->actor, this->actor.speed, this->actor.world.rot.y, this->actor.velocity.y, 0);
+            Actor_PlaySfx(&this->actor, NA_SE_EN_HANDW_RELEASE);
+        } else if (this->timer == 2) {
+            Animation_PlayOnce(&this->skelAnime, &object_wdhand_Anim_000364);
+        }
+    }
+}
+
+s32 EnWdhand_ShrinkLimb(EnWdhand* this, s32 limbIndex) {
+    this->limbScaleFactors[limbIndex] -= 0.1f;
+    if (this->limbScaleFactors[limbIndex] <= 0.05f) {
+        this->limbScaleFactors[limbIndex] = 0.05f;
+        return true;
+    }
+    return false;
+}
+
+void EnWdhand_SetupDie(EnWdhand* this) {
```

Comment:
Nit: `SetupDie`/`Die` is used pretty rarely in the codebase now; it's mostly ported OoT documentation and older actors. The more common one is `SetupDead`/`Dead`; maybe switch to that for consistency?

Reply:
I think `Die` makes more sense for this action. `Dead` conveys a different meaning that I don't think really applies. I probably have this opinion on other areas where `Dead` is used.